### PR TITLE
Remove user facing mentions to Prototype from the new Onboarding flow

### LINKED ISSUE #6298: Update to remove all mentions of Prototype from PO code
### Description

We currently use the word `Prototype` in a few places relating to Progressive Onboarding, a hangover from the Prototype code. We should do a sweep of the code to ensure we're using the correct terminology in all places.

### Acceptance criteria

* Titles and references to PO are up to date and don't mention the prototype.

### PR DESCRIPTION
**Pull request by daquinons**
Fixes #6298 

#### Changes proposed in this Pull Request

This PR removes the "Prototype" mention in the new flow title, and also modifies the path in that new flow from `onboarding-prototype` to `onboarding-flow`.

#### Testing instructions

* Enable Progressive Onboarding to see the new flow
* Onboard a new account
* Make sure the title of the new page is "Onboarding" and not "Onboarding Prototype"
* Check the path of the page and make sure it's `onboarding-flow` and not `onboarding-prototype`.
* Make sure the whole onboarding works as expected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _ 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6332` or branch name `fix/6298-remove-prototype-mention` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6298-remove-prototype-mention) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 088816dd3fd416910c8bf49991c8712f5ae8a992
- Build time: 2023-05-30 07:14:24 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-account.php [hunk 1]
```diff
@@ -1588,12 +1588,12 @@ private function maybe_redirect_to_treatment_onboarding_page() {
...
 
 	/**
-	 * Redirects to the onboarding prototype page if the feature flag is enabled.
+	 * Redirects to the onboarding flow page if the Progressive Onboarding feature flag is enabled.
 	 * Also checks if the server is connect and try to connect it otherwise.
 	 *
 	 * @return void
 	 */
-	private function redirect_to_prototype_onboarding_page() {
+	private function redirect_to_onboarding_flow_page() {
```
Comment (dmallory42) [inline includes/class-wc-payments-account.php:1600]: This function needs to be updated to point the URL to the [new onboarding link](https://github.com/Automattic/woocommerce-payments/blob/c5a2776b934ab597c9e6a3c8b455dc723c448475/includes/class-wc-payments-account.php#L1601-L1601).

I tested updating it and it worked well, so I think that's the only change required. 🙂

**Review by dmallory42 — CHANGES_REQUESTED** [review]
Thanks for working on this, it's looking good!

I spotted one instance where we have a URL in the code which still points to the old prototype (mentioned in the PR comments). Once that's resolved, this will be good to :shipit: !

File: includes/class-wc-payments-account.php (hunk 1 above)
Comment (daquinons) [reply includes/class-wc-payments-account.php:1600]: Oh yes, how I missed it! Changed here https://github.com/Automattic/woocommerce-payments/pull/6332/commits/426d918018f6b4a2c71b52aa65feeca2f2e34030

**Review by dmallory42 — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**